src/transformers/models/dpt/modeling_flax_dpt.py

```diff
@@ -0,0 +1,1236 @@
+
+        outputs = (hidden_states,)
+
+        if output_attentions:
+            outputs += (attn_outputs[1],)
+
+        return outputs
+
+
+# Copied from transformers.models.vit.modeling_flax_vit.FlaxViTIntermediate with ViT->DPT
+class FlaxDPTViTIntermediate(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+
+    def setup(self):
+        self.dense = nn.Dense(
+            self.config.intermediate_size,
+            kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+            dtype=self.dtype,
+        )
+        self.activation = ACT2FN[self.config.hidden_act]
+
+    def __call__(self, hidden_states):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        return hidden_states
+
+
+# DPT reassemble & Fusion
+class FlaxDPTReassembleLayer(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+    factor: int = 1
+    channels: int = None
+
+    def setup(self):
+        # projection
+        self.projection = nn.Conv(
+            self.channels,
+            kernel_size=(1, 1),
+            dtype=self.dtype,
+            kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+            use_bias=True,
+        )
+
+        # up/down sampling depending on factor
+        if self.factor > 1:
+            self.resize = ConvTransposeGradient(
+                self.channels,
+                kernel_size=(self.factor, self.factor),
+                strides=(self.factor, self.factor),
+                use_bias=True,
+            )
+        elif self.factor < 1:
+            # so should downsample
+            self.resize = nn.Conv(
+                self.channels,
+                kernel_size=(3, 3),
+                strides=(int(1 / self.factor), int(1 / self.factor)),
+                dtype=self.dtype,
+                kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+                padding=(1, 1),
+            )
+
+    def __call__(self, hidden_state):
+        hidden_state = self.projection(hidden_state)
+        if self.factor != 1:
+            hidden_state = self.resize(hidden_state)
+        return hidden_state
+
+
+class FlaxDPTReassembleLayerCollection(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.layers = [
+            FlaxDPTReassembleLayer(self.config, factor=factor, channels=self.config.neck_hidden_sizes[i], name=str(i))
+            for i, factor in zip(range(len(self.config.neck_hidden_sizes)), self.config.reassemble_factors)
+        ]
+
+    def __call__(self, x, i):
+        return self.layers[i](x)
+
+
+class FlaxDPTReadoutProjectSequentialCollectionLayer(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.dense = nn.Dense(self.config.hidden_size, name="0")
```

Comment:
Do we need to define name? Will be overwritten by `name=str(i)` in L337

Reply:
I think that I had to put it because we need the dense layer to have the key value `0` to match the state dict of Pytorch.
Since in pytorch the `FlaxDPTReadoutProjectSequentialCollectionLayer` are `nn.Sequential` layers the first layer will have the key `0` -> see [this](https://github.com/younesbelkada/transformers/blob/010a915ce62bd1a3adb61d0eab2f0acca436b1c0/src/transformers/models/dpt/modeling_dpt.py#L438) line where it happens in Pytorch. That is also why the class is called `FlaxDPTReadoutProjectSequentialCollectionLayer`. 
Will add a comment to explain why we need this name

Reply:
Added more comments at: a40f21dfe73079f26c5afccccad1e1cf14d15ac8